homeassistant/components/onvif/camera.py

```diff
@@ -103,59 +125,130 @@ def __init__(self, hass, config):
         self._name = config.get(CONF_NAME)
         self._ffmpeg_arguments = config.get(CONF_EXTRA_ARGUMENTS)
         self._profile_index = config.get(CONF_PROFILE)
+        self._ptz_service = None
         self._input = None
-        self._media_service = \
-            onvif.ONVIFService('http://{}:{}/onvif/device_service'.format(
-                self._host, self._port),
-                               self._username, self._password,
-                               '{}/wsdl/media.wsdl'.format(os.path.dirname(
-                                   onvif.__file__)))
-
-        self._ptz_service = \
-            onvif.ONVIFService('http://{}:{}/onvif/device_service'.format(
-                self._host, self._port),
-                               self._username, self._password,
-                               '{}/wsdl/ptz.wsdl'.format(os.path.dirname(
-                                   onvif.__file__)))
-
-    def obtain_input_uri(self):
+
+        _LOGGER.debug("Setting up the ONVIF camera device @ '%s:%s'",
+                      self._host,
+                      self._port)
+
+        self._camera = ONVIFCamera(self._host,
+                                   self._port,
+                                   self._username,
+                                   self._password,
+                                   '{}/wsdl/'
+                                   .format(os.path.dirname(onvif.__file__)))
+
+    async def async_initialize(self):
+        """
+        Initialize the camera.
+
+        Initializes the camera by obtaining the input uri and connecting to
+        the camera. Also retrieves the ONVIF profiles.
+        """
+        _LOGGER.debug("Updating service addresses")
+
+        await self._camera.update_xaddrs()
+
+        _LOGGER.debug("Setting up the ONVIF device management service")
+
+        devicemgmt = self._camera.create_devicemgmt_service()
+
+        _LOGGER.debug("Retrieving current camera date/time")
+
+        system_date = dt.datetime.utcnow()
```

Comment:
Let's use our built-in utcnow function. 

```python
from homeassistant.util import dt as dt_util
```

```suggestion
        system_date = dt_util.utcnow()
```

Reply:
Fixed. I am assuming that we can still use this code:

```
        cam_date = dt.datetime(cdate.Date.Year, cdate.Date.Month,
                               cdate.Date.Day, cdate.Time.Hour,
                               cdate.Time.Minute, cdate.Time.Second)
```

If that's not the case, please let me know.